sdk/attestation/azure-security-attestation/azure/security/attestation/_models.py

```diff
@@ -69,242 +79,244 @@ class PolicyCertificatesModificationResult(object):
+    or :meth:`azure.security.attestation.AttestationAdministrationClient.reset_policy`
+    API call.
+
+    The `AttestationPolicyResult` class is returned as the body of an attestation token from
     the attestation service. It can be used to ensure that the attestation service
     received the policy object sent from the client without alteration.
-    
-    :param policy_resolution: The result of the policy set or
-        reset call.
+
+    :param policy_resolution: The result of the policy set or reset call.
     :type policy_resolution: azure.security.attestation.PolicyModification
     :param policy_signer: If the call to `set_policy` or `reset_policy`
-        had a :class:`AttestationSigningKey` parameter, this will be the certificate
+        had a `signing_certificate` parameter, this will be the certificate
         which was specified in this parameter.
     :type policy_signer: azure.security.attestation.AttestationSigner
     :param str policy_token_hash: The hash of the complete JSON Web Signature
         presented to the `set_policy` or `reset_policy` API.
 
     """
+
     def __init__(self, policy_resolution, policy_signer, policy_token_hash):
-        #type:(PolicyModification, JSONWebKey, str) -> None
+        # type: (PolicyModification, JSONWebKey, str) -> None
         self.policy_resolution = policy_resolution
         self.policy_signer = AttestationSigner._from_generated(policy_signer)
         self.policy_token_hash = policy_token_hash
 
     @classmethod
     def _from_generated(cls, generated):
-        #type:(GeneratedPolicyResult)->PolicyResult
+        # type: (GeneratedPolicyResult, str) -> AttestationPolicyResult
+        # If we have a generated policy result or policy text, return that.
         if not generated:
-            return cls(None, None, None)
-        return cls(generated.policy_resolution, generated.policy_signer, generated.policy_token_hash)
+            return None
+        return AttestationPolicyResult(
+            generated.policy_resolution,
+            generated.policy_signer,
+            generated.policy_token_hash,
+        )
+
 
-class AttestationResult(object):
-    """ An AttestationResult represents the claims returned from the attestation
+class AttestationResult(object):  # pylint: disable=too-many-instance-attributes
+    """An AttestationResult represents the claims returned from the attestation
     service as a result of a call to
     :meth:`azure.security.attestation.AttestationClient.attest_sgx`, or :meth:`AttestationClient.attest_open_enclave`.
 
+    :keyword str issuer: Entity which issued the attestation token.
+    :keyword unique_identifier: Unique identifier for the token.
+    :paramtype unique_identifier: str or None
+    :keyword nonce: Returns the input `nonce` attribute passed to the `attest` API.
+    :paramtype nonce: str or None
+    :keyword str version: Version of the token. Must be "1.0"
```

Comment:
Interesting. If it's an optional parameter but only one value is valid for now, does it make sense to expose it?

Reply:
I think there must be a mismatch between github's view of your comment and where you made the comment. I'm not sure I understand the context of your question.

In general, anything that's returned from the service is exposed to the client. 

If you're referring to the `nonce` attribute, it's an unfortunate screw-up on our part. The `nonce` attribute was defined and is supported by the service in all API versions, but it didn't make the approved swagger. Fixing this is a breaking change and we're not bringing up a breaking change to our swagger for this. Newer versions of the API will add support for this attribute, at which point clients can specify it in their requests.

But there field is in the swagger in the output.

Reply:
I think mccoy is referring to the `:keyword str version: Version of the token. Must be "1.0"` line. What is the purpose of an optional keyword with only one option?